test/e2e_node/remote/ssh.go

```diff
@@ -70,7 +70,8 @@ func GetHostnameOrIP(hostname string) string {
 		host = ip
 	}
 
-	if *sshUser == "" {
+	// Do no ignore the KUBE_SSH_USER if it's set explicitly
+	if os.Getenv("KUBE_SSH_USER") != "" {
 		*sshUser = os.Getenv("KUBE_SSH_USER")
```

Comment:
as discussed above the flag should take precedence over the environment variable.

please change the logic to something like:

```
var user string
if *sshUser != "" {
	user = *sshUser
	goto add_user
}
if os.Getenv("KUBE_SSH_USER") != "" {
	user = os.Getenv("KUBE_SSH_USER")
	goto add_user
}
return host
add_user:
	return fmt.Sprintf("%s@%s", user, host)
```